### DISCUSSION TITLE
Data type (Bit / INTEGER) incompatibility between schema and file in MysqltoGCSOperator

### DISCUSSION THREAD

**Discussion by Walid-peach** [apache/airflow#19451] [General]
### Apache Airflow Provider(s)

google, mysql

### Apache Airflow version

2.1.4 (latest released)

### Operating System

ubuntu 20.04

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by uranusjr** [comment]
Can you elaborate?

**Comment by Walid-peach** [comment]
in [class MySQLToGCSOperator](https://airflow.apache.org/docs/apache-airflow-providers-google/stable/_modules/airflow/providers/google/cloud/transfers/mysql_to_gcs.html#MySQLToGCSOperator)
the definition  `type_map = {
        FIELD_TYPE.BIT: '**INTEGER**', ....` is incompatible with the function `def convert_type(self, value, schema_type: str):
          ......
          if schema_type == "INTEGER":
                value = int.from_bytes(value, "big")
            else:
                value = base64.standard_b64encode(value).decode('ascii')` 
which lead us to the error 400 when we want to transfer the table from GCS to BigQuery
`google.api_core.exceptions.BadRequest: 400 Error while reading data, error message: JSON table encountered too many errors, giving up. Rows: 1; errors: 1. Please look into the errors[] collection for more details.`

**Comment by uranusjr** [comment]
Could you please try to edit your message so it can be understood.

**Comment by eladkal** [comment]
@Walid-peach if you found the issue. Can you please open a PR to fix it?

**Comment by potiuk** [comment]
Until there is a proposal of improvement (ideally as PR) I move it into discussion
